tools/management/commands/get_gtp_ligands.py

```diff
@@ -0,0 +1,491 @@
+        self.analyse_rows()
+        try:
+            print('Updatind ligand data from GuideToPharma')
+
+            print('COMPLETED updating GuideToPharma Data')
+        except Exception as msg:
+            print('--error--', msg, '\n')
+            self.logger.info("The error appeared in def handle")
+
+# pylint: disable=R0201
+    def purge_bias_data(self):
+        print("# Purging data")
+        delete_bias_experiment = GTP_endogenous_ligand.objects.all()
+        delete_bias_experiment.delete()
+
+    def analyse_rows(self):
+        """
+        Fetch data to models
+        Saves to DB
+        """
+        print('---Starting---\n')
+        print("\n#1 Get GPCR ids from target families")
+        target_list = self.get_gpcrs()
+        print("\n#2 Get Endogeneous ligands")
+        self.get_endogenous(target_list)
+        print("\n#3 Get endogenous data from GPCRDb")
+        endogenous_ligands_from_db = self.endogenous_ligands_from_db()
+        print("\n#4 Convert_query_to_dict" )
+        endogenous_list = self.convert_query_to_dict(endogenous_ligands_from_db)
+        print("\n#5 Combine same assays")
+        combined_assays = self.process_assays(endogenous_list)
+        print("\n#6 Prepare to calculate averages and save")
+        prepared_data = self.prepare_calculate_averages_and_save(combined_assays)
+        print("\n#7 Calculate averages")
+        averaged_list = self.calculate_averages(prepared_data)
+        print("\n#8 Save")
+        self.save_data(averaged_list)
+        print('\n\n---Finished---')
+
+    def get_endogenous(self, targets):
+        for target in targets:
+            protein = self.fetch_protein(target)
+            response = requests.get(
+                "https://www.guidetopharmacology.org/services/targets/" + str(target) + "/naturalLigands")
+            if response.status_code == 200:
+                data = response.json()
+                for i in data:
+                    # try:
+                    # import pdb; pdb.set_trace()
+                    ligand_name = str()
+                    try:
+                        ligand_name = self.ligand_cache[i['targetId']]
+                    except:
+                        ligand_name = ""
+
+                    ligand = self.fetch_ligand(
+                        i['ligandId'], i['type'], ligand_name)
+                    if ligand and protein:
+                        # if target == 1:
+
+                        self.get_ligand_interactions(ligand=ligand, ligand_id_gtp=data[0]['ligandId'], ligand_type=data[0]['type'],receptor=protein)
+                    else:
+                        pass
+                    # except:
+                    #     pass
+
+    def get_ligand_interactions(self, ligand, ligand_id_gtp, ligand_type, receptor):
+        response = requests.get(
+            "https://www.guidetopharmacology.org/services/ligands/" + str(ligand_id_gtp) + "/interactions")
+        if response.status_code == 200:
```

Comment:
You have to filter the activities based on the receptor as this request returns all "interactions" for that specific ligand. Use the GtP target ID to do this

Reply:
that makes sense...